students/roslyn_m/lesson05/MailroomPart3.py

```diff
@@ -0,0 +1,154 @@
+            person["Qty"] = person["Qty"] + 1
+            person["Avg"] = round(person["Total"] / person["Qty"], 2)
+            existing = True
+            person_to_thank = person
+    if existing != True:
+        try:
+            donation_amt = float(input("Please enter the donation amount: "))
+        except ValueError:
+            print("Your input was not valid!")
+            donation_amt = float(input("Please enter the donation amount: "))
+        person_add = {"Name": user_input.title(), "Total": donation_amt, "Qty": 1, "Avg": donation_amt}
+        donor_all.append(person_add)
+        person_to_thank = person_add
+    print("The following thank you note will be emailed to the donor: ")
+    print(
+        '"Dear {}, \n Thank you so much for your gracious donation of ${:.2f}. We are so thankful for your strong support!! \nCheers,\nRoslyn Melookaran"'.format(
+            person_to_thank["Name"], donation_amt))
+
+
+def create_report(donor_all):
+    """ Create a report of donors to show user.
+                :param: donor_all (list of list)
+                :return: nothing
+                """
+    title = ["Donor Name", "Total Given", "Num Gifts", "Average Gift"]
+    print(f'{title[0]:20}|{title[1]:^12}|{title[2]:^10}|{title[3]:^14}|')
+    print("-" * 60)
+    for person in donor_all:
+        print(f'{person["Name"]:20} ${person["Total"]:^12}{person["Qty"]:9}   ${person["Avg"]:^12}')
+    print("\n")
+
+
+def order_list(donor_all):
+    """ Sort list of donors by donation amount total.
+            :param: donor_all (list of list)
+            :return: sorted_list (list of lists)
+            """
+    sorted_list = sorted(donor_all, key=lambda x: -(x["Total"]))
+    return sorted_list
+
+
+def send_all_thank_you(donor_all):
+    """ Function will print a .txt file thank you note for each donor in list
+            :param: donor_all (list of lists)
+            :return: nothing
+            """
+    for person in donor_all:
+        string = "Dear {Name}, \n\n \t Thank you so much for your support! You have donated a total of ${Total}, which is so very appreciated. We hope to have your continued support through these times. \n\nThanks, \nRoslyn".format(**person)
+        print(string)
+        filename = person["Name"]
+        filename = filename.replace(" ", "")
+        filename = filename + ".txt"
+        with open(filename, 'w') as f:
+            f.write(string)
+    return
+
+
+# ----------Variables----------#
+person1 = {"Name": 'William Gates', "Total": 653784.49, "Qty": 2, "Avg": 326892.24}
+person2 = {"Name": 'Mark Zuckerberg', "Total": 16396.10, "Qty": 3, "Avg": 5465.37}
+person3 = {"Name": 'Jeff Bezos', "Total": 877.33, "Qty": 1, "Avg": 877.33}
+person4 = {"Name": 'Paul Allen', "Total": 708.42, "Qty": 3, "Avg": 236.14}
+person5 = {"Name": 'John Smith', "Total": 899.98, "Qty": 5, "Avg": 399.21}
+donor_list = [person1, person2, person3, person4, person5]
```

Comment:
I should have caught this last version, but at this point the main structure of your database should be using a dictionary.  This is a list of dictionaries.  You'll want something like:  donor_dict = {'person1': [100.00, 200.50], 'person2: [50.25, 60.00]
etc. so each key is a donor and the values are a list of their donations.